Sell/OrderingJourney/OrderingJourneyView.swift

```diff
@@ -25,29 +25,50 @@ public struct OrderingJourneyView: View {
     public let urlRequest: URLRequest
     public let colorStore: ColorStorable
     private let scheduler: DispatchQueueAnyScheduler
+    private var javascriptRunDetails: JavascriptRunDetails?
+    public let applicationURLOpener: ApplicationURLOpening
+    private var webViewWrapper: WebViewWrapper!
     let inspection = Inspection<Self>()
 
-    init(urlString: String, colorStore: ColorStorable, scheduler: DispatchQueueAnyScheduler) {
+    init(urlString: String,
+         colorStore: ColorStorable,
+         scheduler: DispatchQueueAnyScheduler,
+         javascriptRunDetails: JavascriptRunDetails?,
+         applicationURLOpener: ApplicationURLOpening) {
         let url = URL(string: urlString) ?? URL(fileURLWithPath: "")
-        self.urlRequest = URLRequest(url: url)
+        let urlRequest = URLRequest(url: url)
+        self.urlRequest = urlRequest
         self.colorStore = colorStore
         self.scheduler = scheduler
+        self.javascriptRunDetails = javascriptRunDetails
+        self.applicationURLOpener = applicationURLOpener
     }
 
-    public init(urlString: String, colorStore: ColorStorable) {
+    public init(urlString: String,
+                colorStore: ColorStorable,
+                javascriptRunDetails: JavascriptRunDetails?,
+                applicationURLOpener: ApplicationURLOpening) {
         let url = URL(string: urlString) ?? URL(fileURLWithPath: "")
         self.urlRequest = URLRequest(url: url)
         self.colorStore = colorStore
         self.scheduler = .main
+        self.javascriptRunDetails = javascriptRunDetails
+        self.applicationURLOpener = applicationURLOpener
+    }
```

Comment:
May I know what's the purpose for having two init function please? Thank you.

Reply:
I'm actually not sure. Mickey made this file originally and I just adapted it. It might be so it can be used with SwiftUI or normal view controllers...